### PR TITLE
Add support for variable font hinting

### PR DESCRIPTION
**Pull request by madig**
TODO:
- [ ] Compile `cvar` table
- [x] Remove `TSIC` table on `compile --ship`
- [ ] Test

### CODE REVIEW

**Comment by madig** [comment]
Hm. VTT seems to maintain the `cvar` table itself, so maybe there doesn't need to be any compilation.

**Comment by madig** [comment]
Plot twist: VTT manages both `TSIC` and `cvar` tables and the `TSIC` data is not a strict superset of the `cvar` data. Rounding is also different from fontTools's `otRound`. Conclusion: don't try to compile `TSIC`, just dump and load both tables.

**Comment by madig** [comment]
Another plot twist: I can't read `cvar` from a TTX dump without having read `cvt ` first. Or compiled?

File: src/vttLib/transfer.py [hunk 1]
```diff
@@ -10,29 +10,43 @@
...
 
-def dump_to_file(font: fontTools.ttLib.TTFont, path: os.PathLike):
+def dump_to_file(font: fontTools.ttLib.TTFont, path: os.PathLike) -> None:
     """Dump relevant VTT data to a file.
 
     Relevant are:
     - TSI1 holds the assembly code
     - TSI3 holds the VTT code
     - TSI5 holds glyph group information
+    - TSIC holds data for the cvar table BUT IS NOT A STRICT SUPERSET
```
Comment (belluzj) [inline src/vttLib/transfer.py]: Do you have an example of what is in cvar but not TSIC, for reference in the future?

**Review by belluzj — APPROVED** [review]

File: src/vttLib/transfer.py (hunk 1 above)
Comment (madig) [reply src/vttLib/transfer.py]: What I found with a font is that there was data in cvar that was not in TSIC (deleting cvar and recompiling TSIC in VTT compiled a smaller cvar). So theoretically both can contain the same data, but as of VTT 6.33 Beta they probably don't?

**Pull request merged**